wsdl-core/src/main/java/io/ballerina/wsdl/core/SchemaManager.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.wsdl.core;
+
+import org.apache.ws.commons.schema.XmlSchema;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SchemaManager {
+
+    private static final SchemaManager instance = new SchemaManager();
+
+    private Map<String, XmlSchema> targetNSToSchema;
+
+    private SchemaManager() {
+        targetNSToSchema = new HashMap<>();
+    }
+
+    public static SchemaManager getInstance() {
+        return instance;
+    }
+
+    public void initializeSchemas(Map<String, XmlSchema> targetNSToSchema) {
+        this.targetNSToSchema = targetNSToSchema;
+    }
```

Comment:
We don't need this function to initialize the `SchemeManager` right ? We can do this within the constructor itself.

Reply:
This is a singleton, Constructor is private